by default, availability-disabled headers indicate the supplied libcxx contains
a verbose abort function. the libcxx versions that are installed by macports-libcxx 
do not currently have such a function, so links fail.

alter the default availability header to indicate no such function is supplied

see: https://trac.macports.org/ticket/68640

--- a/libcxx/include/__availability.orig	2024-04-07 10:11:44.000000000 -0700
+++ b/libcxx/include/__availability	2024-04-07 10:12:25.000000000 -0700
@@ -159,7 +159,7 @@
     // This controls whether the library claims to provide a default verbose
     // termination function, and consequently whether the headers will try
     // to use it when the mechanism isn't overriden at compile-time.
-// #   define _LIBCPP_HAS_NO_VERBOSE_ABORT_IN_LIBRARY
+#   define _LIBCPP_HAS_NO_VERBOSE_ABORT_IN_LIBRARY
 
 #elif defined(__APPLE__)
 
